### PR TITLE
feat: Update CODEOWNERS for Tuxedo Pulse Laptops

### PR DESCRIPTION
**Pull request by gabyx**
###### Description of changes

Add codeowners for reviews in Tuxedo Pulse Laptops.

###### Things done

- [x] Tested the changes in your own NixOS Configuration
- [x] Tested the changes end-to-end by using your fork of `nixos-hardware` and
      importing it via `<nixos-hardware>` or Flake input

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -11,3 +11,4 @@ purism/librem/13v3 @yegortimoshenko
 system76/darp6 @khumba
 framework @emiller88
 tuxedo/pulse/15/gen2 @trueNAHO
+tuxedo/pulse/14/gen3 @gabyx @britter @trueNAHO
```
Comment (Mic92) [inline CODEOWNERS:14]: Do code owners actually work, if the owners are not committers on this repository?

Comment (gabyx) [reply CODEOWNERS:14]: Dont know :)

Comment (Mic92) [reply CODEOWNERS:14]: We can probably keep these lines, but github is throwing warnings for all non-nixpkgs committers.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**